client/options.go

```diff
@@ -153,7 +153,7 @@ func WithTLSClientConfig(cacertPath, certPath, keyPath string) Opt {
 func WithVersion(version string) Opt {
 	return func(c *Client) error {
 		if version != "" {
-			c.version = version
+			c.setVersion(version)
```

Comment:
`WithVersion()` is used when constructing a new client, and those options are always applied sequentially, so looks like we don't need to take a lock here.